codegen/context.go

```diff
@@ -191,3 +194,27 @@ func DefaultPlatform(ctx context.Context) specs.Platform {
 	}
 	return platform
 }
+
+type DockerAPIClient struct {
+	dockerclient.APIClient
+	Auth imagetools.Auth
+	Moby bool
+	err  error
+}
+
+func WithDockerAPI(ctx context.Context, cln dockerclient.APIClient, auth imagetools.Auth, err error, moby bool) context.Context {
+	return context.WithValue(ctx, dockerAPIKey{}, &DockerAPIClient{
+		APIClient: cln,
+		Auth:      auth,
+		Moby:      moby,
+		err:       err,
+	})
+}
+
+func DockerAPI(ctx context.Context) (*DockerAPIClient, error) {
```

Comment:
Any particular reason to return a pointer here?

Reply:
No reason, originally I returned `nil` in the error case.